### DISCUSSION TITLE
Can I load schema from a npm package?

### DISCUSSION THREAD

**Discussion by borekb** [dotansimha/graphql-code-generator#5485] [Q&A]
Is something like this possible?

```yaml
schema: '@my/package/schema.graphql'
```

`@my/package` is installed from npm and the `.graphql` file can be found in `node_modules/@my/package`.

I think it's currently not supported directly, maybe I could use a workaround with unpkg.com or similar? (In my specific case, the package is not published to npm yet; it's a local package linked to my main project via Yarn workspaces.)

**Comment by ardatan** [comment]
@borekb Could you try the following;
```yml
schema: module:@my/package/schema.graphql
```

**Reply by borekb** [reply]
That doesn't work for me, unfortunately.

**UPDATE**: I tried both "simple" structure with `node_modules/@my/package` directly in my main project, and the Yarn workspaces setup where all the `@my/*` packages are hoisted to root node_modules. Neither worked.

**Reply by borekb** [reply]
@ardatan Can you point me to the part of the source code where tricks like `module:` happen? 🙏

**Reply by ardatan** [reply]
@borekb Here;
https://github.com/ardatan/graphql-tools/blob/master/packages/loaders/module/src/index.ts

**Reply by borekb** [reply]
I have a hard time figuring out how to possibly use this. I always get a generic error like:

```
Failed to load schema from module:@my/graphql#typeDefs:


Unable to find any GraphQL type definitions for the following pointers:

    - module:@my/graphql#typeDefs

  Error:
Unable to find any GraphQL type definitions for the following pointers:

    - module:@my/graphql#typeDefs

at prepareResult (/Users/borekb/dev/my/project/packages/dist/load/src/load-typedefs.js:70:15)
at loadTypedefs (/Users/borekb/dev/my/project/packages/dist/load/src/load-typedefs.js:35:12)
at Object.loadSchema (/Users/borekb/dev/my/project/packages/dist/load/src/schema.js:11:21)
at loadSchema (/Users/borekb/dev/my/project/packages/node_modules/@graphql-codegen/cli/bin.js:452:24)
at /Users/borekb/dev/my/project/packages/node_modules/@graphql-codegen/cli/bin.js:872:55
at Task.task (/Users/borekb/dev/my/project/packages/node_modules/@graphql-codegen/cli/bin.js:725:17)

  GraphQL Code Generator supports:
    - ES Modules and CommonJS exports (export as default or named export "schema")
    - Introspection JSON File
    - URL of GraphQL endpoint
    - Multiple files with type definitions (glob expression)
    - String in config file

  Try to use one of above options and run codegen again.
Error: Failed to load schema
  at loadSchema (/Users/borekb/dev/my/project/packages/node_modules/@graphql-codegen/cli/bin.js:460:15)
Error: Failed to load schema
  at loadSchema (/Users/borekb/dev/my/project/packages/node_modules/@graphql-codegen/cli/bin.js:460:15)
```

This is my `codegen.yml`:

```yaml
generates:
  graphql-api/schema.graphql:
    schema: 'module:@shoptet/graphql#typeDefs'
    plugins:
      - schema-ast
```

I'm not even sure how the ModuleLoader should be installed – I think it has to be installed explicitly (I couldn't find it in `node_modules` with the default codegen installation) but even with that, I think the code in the module loader is no executing.

How does an example setup with a module loader look like?
